File: .tmp_py4013/batch_04_sources/2509_cycle_length_queries_in_a_tree.js
// LeetCode 2509 - Cycle Length Queries in a Tree
// https://leetcode.com/problems/cycle-length-queries-in-a-tree/

/**
 * @param {number} n
 * @param {number[][]} queries
 * @return {number[]}
 */
var cycleLengthQueries = function(n, queries) {
    const ans = Array(queries.length);
    for (let i = 0; i < queries.length; i++) {
        let a = queries[i][0], b = queries[i][1], steps = 0;
        while (a !== b) {
            if (a > b) a = Math.floor(a / 2);
            else b = Math.floor(b / 2);
            steps++;
        }
        ans[i] = steps + 1;
    }
    return ans;
};

File: .tmp_rs4013/rust_check_batch08/2509_cycle_length_queries_in_a_tree.rs
struct Solution;
// LeetCode 2509 - Cycle Length Queries in a Tree
// https://leetcode.com/problems/cycle-length-queries-in-a-tree/

impl Solution {
    pub fn cycle_length_queries(_n: i32, queries: Vec<Vec<i32>>) -> Vec<i32> {
        let mut ans = vec![0; queries.len()];
        for (i, q) in queries.iter().enumerate() {
            let mut a = q[0];
            let mut b = q[1];
            let mut steps = 0;
            while a != b {
                if a > b {
                    a /= 2;
                } else {
                    b /= 2;
                }
                steps += 1;
            }
            ans[i] = steps + 1;
        }
        ans
    }
}

fn main() {}
